### PR TITLE
Handle stop message received before ACS query completion

### PR DESCRIPTION
**Pull request by shayne-fletcher**
Handle the case a `Stop` message is seen by a `TriggerActor` before its ACS query has completed.

I'm sorry that I don't have a test for this. How to write one is a little beyond me just for the moment - I'm open to ideas!

### CODE REVIEW

File: triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala [hunk 1]
```diff
@@ -96,24 +96,33 @@ object TriggerActor {
...
+        Behaviors.receiveMessagePartial[Message] {
+          case QueryACSFailed(cause) => throw new RuntimeException("ACS query failed", cause)
+          case QueriedACS(runner, acs, offset) =>
+            val (killSwitch, trigger) = runner.runWithACS(
+              acs,
+              offset,
+              msgFlow = KillSwitches.single[TriggerMsg],
+            )
+            // TODO If we are stopped we will end up causing the future to complete which will trigger
+            // a message that is sent to a now terminated actor. We should fix this somehow™.
```
Comment (rohanjr) [inline triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala:111]: Is this TODO what you are fixing?

Comment (shayne-fletcher) [reply triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala:111]: No, the TODO that was on line 100.

File: triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala [hunk 2]
```diff
@@ -96,24 +96,33 @@ object TriggerActor {
...
-          }
-          running(killSwitch)
-      }
+      // Waiting for the ACS query to finish so we can build the
+      // initial state.
+      def queryingACS(wasStopped: Boolean): Behaviors.Receive[Message] =
+        Behaviors.receiveMessagePartial[Message] {
+          case QueryACSFailed(cause) => throw new RuntimeException("ACS query failed", cause)
+          case QueriedACS(runner, acs, offset) =>
+            val (killSwitch, trigger) = runner.runWithACS(
```
Comment (cocreature) [inline triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala:105]: This doesn’t look right to me. If we got stopped we don’t want to spawn the trigger here only to then tear it down once we receive the stop message. This should just not start in the first place.

Comment (shayne-fletcher) [reply triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala:105]: Ok. I see what you mean. I'll adjust it accordingly. Thanks!

**Comment by shayne-fletcher** [comment]
I'm a gonna land this but please don't be discouraged from making further comments @cocreature if you come back to this!

**Comment by shayne-fletcher** [comment]
> Thanks! LGTM now. I also don’t have a good idea for how to test this so happy to merge it as is for now.

Oops. One of these days I will learn how to read! 🤦

**Pull request merged**